### PR TITLE
Add metrics examples

### PR DESCRIPTION
**Pull request by ezhong2**
Adds examples for metrics operations 
- Example for `createMetricSource` mutation
- Example for `createMetricDefinition` mutation
- Example for `getMetricDefinition` query (to retrieve all metric values for a given metric definition) 
- Sample bash script for adding a metric to components using `createMetricSource`

### CODE REVIEW

File: snippets/graphql/create-metric-definition/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}
+```
+
+### Query Variables
+
+```
+{
+  "input": {
+    "cloudId": "<cloud_id>",
+    "name": "<name of your metric definition>"
```
Comment (eserranovalle) [inline snippets/graphql/create-metric-definition/README.md:40]: Worth adding a `,` after the `...metric definition>"`

File: snippets/graphql/create-metric-definition/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    "format": {
+      "suffix": {
+        "suffix": <the units of your metric>
```
Comment (eserranovalle) [inline snippets/graphql/create-metric-definition/README.md:43]: Minor: for consistency, putting `"` around the placeholder would make sense. -> `"<the units of your metric>"`

**Review by eserranovalle — APPROVED** [review]
Two very minor changes, but otherwise looks good. This is really useful, @ezhong2 !

**Review by eserranovalle — APPROVED** [review]

File: snippets/graphql/add-metric-to-components/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+This following steps can be used to add a metric for a subset or all components.
+
+1. Create or retrieve a metric definition. Follow [createMetricDefintion](/snippets/graphql/create-metric-definitions/README.md) to create a metric definition and save the `metric_definition_id`. To retrieve an existing metric definiton ID, follow [getMetricDefinitions](/snippets/graphql/get-metric-definitions/README.md) to retrieve a list of metric definitions on your site, or to add a predefined metric, grab the ID from [this list](https://developer.atlassian.com/cloud/compass/components/available-predefined-metrics/).
```
Comment (sbland2) [inline snippets/graphql/add-metric-to-components/README.md:3]: [nit] typo on `createMetricDefinition`

File: snippets/graphql/add-metric-to-components/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+This following steps can be used to add a metric for a subset or all components.
+
+1. Create or retrieve a metric definition. Follow [createMetricDefintion](/snippets/graphql/create-metric-definitions/README.md) to create a metric definition and save the `metric_definition_id`. To retrieve an existing metric definiton ID, follow [getMetricDefinitions](/snippets/graphql/get-metric-definitions/README.md) to retrieve a list of metric definitions on your site, or to add a predefined metric, grab the ID from [this list](https://developer.atlassian.com/cloud/compass/components/available-predefined-metrics/). 
+
+2. Retrieve the subset of components you would like to apply the metric definition to. Follow [searchComponents](/snippets/graphql/search-components/README.md) to search components based on various filters and retrieve `component_ids`. 
+
+3. For each component, call [createMetricSource](/snippets/graphql/create-metric-source/README.md) with the `metric_definition_id` to apply the metric definition. The `externalMetricSourceId` is an identifier of your metric source in an external system, for example, a Bitbucket repository UUID.
```
Comment (sbland2) [inline snippets/graphql/add-metric-to-components/README.md:7]: "A unique identifier of your metric source in an external system" maybe?

Comment (sbland2) [reply snippets/graphql/add-metric-to-components/README.md:7]: It being unique is something to drive home

File: snippets/graphql/create-metric-definition/README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# How to use this
+
+This query will create a metric defintion on your site.
```
Comment (kairophan) [inline snippets/graphql/create-metric-definition/README.md:3]: nit: also typo here on definition

File: snippets/graphql/get-metric-values-for-metric-definition/README.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
+This query can be used to get all metric values for metric sources with a specified metric definition.
```
Comment (kairophan) [inline snippets/graphql/get-metric-values-for-metric-definition/README.md:1]: instead of all, maybe say "paginate-able"?

**Review by kairophan — APPROVED** [review]

**Pull request merged**